pvlib/tests/iotools/test_pvgis.py

```diff
@@ -8,10 +8,9 @@
 import pytest
 import requests
 from pvlib.iotools import get_pvgis_tmy
+from conftest import data_dir
 
-TESTS = Path(__file__).parent
-PROJECT = TESTS.parent
-DATA = PROJECT / 'data'
+DATA = Path(data_dir)
```

Comment:
Would it be better (or not) to use the `os.path.join` pattern here instead of importing and using `Path`? I don't have a strong preference except for consistency.

Reply:
Good point. I was just playing around with Python-3, because always wanted to start using `Pathlib` and Python-3 only features more. I think `os` is an old library, so maybe it's slightly more Pythonic to use `Pathlib`, but I'm not sure. I'm open with doing whatever for consistency, but here are a few posts in favor of `Pathlib`:
* [**Why you should be using pathlib**](https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/)
* [**No really, pathlib is great**](https://treyhunner.com/2019/01/no-really-pathlib-is-great/)
* [**Python 3's pathlib Module: Taming the File System**](https://realpython.com/python-pathlib/)
* [**Is pathlib a viable replacement for os.path?**](https://python-forum.io/Thread-Is-pathlib-a-viable-replacement-for-os-path)
* [**Easier Python paths with pathlib**](https://www.linuxjournal.com/content/easier-python-paths-pathlib)
* [**Why use the Pathlib Module?**](https://stackabuse.com/introduction-to-the-python-pathlib-module/#whyusethepathlibmodule)
* etc. I just [Googled `Pathlib` vs. `os`](https://www.google.com/search?q=os.path.join+vs+pathlib)


There is one solid benefit to using `Path` objects over os, which is that it has the `open` method built in, so the old pattern:

    with open(os.path.join('path', 'to', 'file')) as f:
        # do stuff with f

becomes

    with (Path('path') / 'to' / 'file').open() as f
        # do stuff with f

which seems a little clearer perhaps? You can see this in action right here:
https://github.com/pvlib/pvlib-python/blob/d635c09123319c68182898001712377089199744/pvlib/test/test_pvgis.py#L61-L64

Pretty cool, no?

Reply:
what about defining `data_dir` and `test_dir` as `Path` objects in `conftest`? Seems like that would avoid a lot of `os` and `pathlib` imports in individual test modules.